commons/zenoh-protocol/src/core/endpoint.rs

```diff
@@ -443,10 +446,29 @@ impl fmt::Debug for ConfigMut<'_> {
 
 /// A string that respects the [`EndPoint`] canon form: `<locator>[#<config>]`.
 ///
-/// `<locator>` is a valid [`Locator`] and `<config>` is of the form `<key1>=<value1>;...;<keyN>=<valueN>` where keys are alphabetically sorted.
-/// `<config>` is optional and can be provided to configure some aspectes for an [`EndPoint`], e.g. the interface to listen on or connect to.
+/// `<locator>` is a valid [`Locator`] and `<config>` is of the form
+/// `<key1>=<value1>;...;<keyN>=<valueN>` where keys are alphabetically sorted. `<config>` is
+/// optional and can be provided to configure some aspects for an [`EndPoint`], e.g. the interface
+/// to listen on or connect to.
 ///
 /// A full [`EndPoint`] string is hence in the form of `<proto>/<address>[?<metadata>][#config]`.
+///
+/// ## Metadata
+///
+/// - **`priorities`**: a range bounded inclusively below and above (e.g. `2..=4` signifies
```

Comment:
I'd use some easier format, e.g.
`priorities=1-4` would mean all priorities from `1 to 4 both included`.
`priorities=1` would mean only priority `1`.

This is more efficient in terms of network overhead compared to `2..=4` where we need 3 characters `..=` to express the same meaning of `-`.

Reply:
I understand. Although `-` is a questionable choice since it looks like substraction in this context. I suggest `:` (what Python uses for slicing sequences).

Reply:
Well not so questionable in terms of network overhead. I'd even suggest having something like `prio=1-4` to save a few bytes more during scouting and discovery.